main.go

```diff
@@ -1,426 +0,0 @@
-		}
-
-		configfile := configfile{}
-
-		err = yaml.Unmarshal(b, &configfile)
-		if err != nil {
-			klog.Fatalf("Failed to parse config file content: %v", err)
-		}
-
-		cfg.auth.Authorization = configfile.AuthorizationConfig
-	}
-
-	kubeClient, err := kubernetes.NewForConfig(kcfg)
-	if err != nil {
-		klog.Fatalf("Failed to instantiate Kubernetes client: %v", err)
-	}
-
-	var authenticator authenticator.Request
-	// If OIDC configuration provided, use oidc authenticator
-	if cfg.auth.Authentication.OIDC.IssuerURL != "" {
-		authenticator, err = authn.NewOIDCAuthenticator(cfg.auth.Authentication.OIDC)
-		if err != nil {
-			klog.Fatalf("Failed to instantiate OIDC authenticator: %v", err)
-		}
-	} else {
-		//Use Delegating authenticator
-		klog.Infof("Valid token audiences: %s", strings.Join(cfg.auth.Authentication.Token.Audiences, ", "))
-
-		tokenClient := kubeClient.AuthenticationV1().TokenReviews()
-		delegatingAuthenticator, err := authn.NewDelegatingAuthenticator(tokenClient, cfg.auth.Authentication)
-		if err != nil {
-			klog.Fatalf("Failed to instantiate delegating authenticator: %v", err)
-		}
-
-		go delegatingAuthenticator.Run(1, context.Background().Done())
-		authenticator = delegatingAuthenticator
-	}
-
-	sarClient := kubeClient.AuthorizationV1().SubjectAccessReviews()
-	sarAuthorizer, err := authz.NewSarAuthorizer(sarClient)
-
-	if err != nil {
-		klog.Fatalf("Failed to create sar authorizer: %v", err)
-	}
-
-	staticAuthorizer, err := authz.NewStaticAuthorizer(cfg.auth.Authorization.Static)
-	if err != nil {
-		klog.Fatalf("Failed to create static authorizer: %v", err)
-	}
-
-	authorizer := union.New(
-		staticAuthorizer,
-		sarAuthorizer,
-	)
-
-	auth, err := proxy.New(kubeClient, cfg.auth, authorizer, authenticator)
-
-	if err != nil {
-		klog.Fatalf("Failed to create rbac-proxy: %v", err)
-	}
-
-	upstreamTransport, err := initTransport(cfg.upstreamCAFile)
-	if err != nil {
-		klog.Fatalf("Failed to set up upstream TLS connection: %v", err)
-	}
-
-	if len(cfg.allowPaths) > 0 && len(cfg.ignorePaths) > 0 {
-		klog.Fatal("Cannot use --allow-paths and --ignore-paths together.")
-	}
-
-	for _, pathAllowed := range cfg.allowPaths {
-		_, err := path.Match(pathAllowed, "")
-		if err != nil {
-			klog.Fatalf("Failed to verify allow path: %s", pathAllowed)
-		}
-	}
-
-	for _, pathIgnored := range cfg.ignorePaths {
-		_, err := path.Match(pathIgnored, "")
-		if err != nil {
-			klog.Fatalf("Failed to verify ignored path: %s", pathIgnored)
-		}
-	}
-
-	proxy := httputil.NewSingleHostReverseProxy(upstreamURL)
-	proxy.Transport = upstreamTransport
-
-	if cfg.upstreamForceH2C {
-		// Force http/2 for connections to the upstream i.e. do not start with HTTP1.1 UPGRADE req to
-		// initialize http/2 session.
-		// See https://github.com/golang/go/issues/14141#issuecomment-219212895 for more context
-		proxy.Transport = &http2.Transport{
-			// Allow http schema. This doesn't automatically disable TLS
-			AllowHTTP: true,
-			// Do disable TLS.
```

Comment:
TODO: pre-acceptance figure out why it's save to disable TLS.  If it isn't safe to disable TLS (seems questionable for a proxy) stop doing it.

Reply:
@deads2k what is kube's stance on supporting h2c? If kube does not explicitly support it, i think this is fine to be removed.

Reply:
(again, a breaking change with announcement necessary if we decide that path)

Reply:
@liggitt @enj transport configuration for proxy

Reply:
By doing some reading on h2c, the code and history digging, I found that this is meant to be used for proxy to upstream communication and not anyone else.
It seems like some folks wanted to protect tiller (which did h2c) and what not.
In combination with your comment here @deads2k, which I agree on, I would revisit the configuration / logic for identity forwarding in such cases.